lib/puppet/provider/ios_interface/cisco_ios.rb

```diff
@@ -0,0 +1,124 @@
+require_relative '../../util/network_device/cisco_ios/device'
+require_relative '../../../puppet_x/puppetlabs/cisco_ios/utility'
+
+# IOS Interface Puppet Provider for Cisco IOS devices
+class Puppet::Provider::IosInterface::CiscoIos
+  def self.commands_hash
+    @commands_hash ||= PuppetX::CiscoIOS::Utility.load_yaml(File.expand_path(__dir__) + '/command.yaml')
+  end
+
+  def self.instances_from_cli(output)
+    new_instance_fields = []
+    output.scan(%r{#{PuppetX::CiscoIOS::Utility.get_instances(commands_hash)}}).each do |raw_instance_fields|
+      new_instance = PuppetX::CiscoIOS::Utility.parse_resource(raw_instance_fields, commands_hash)
+      new_instance_fields << PuppetX::CiscoIOS::Utility.device_safe_instance(new_instance, commands_hash)
+    end
+    cleaned_value = []
+    new_instance_fields.each do |instance|
+      instance[:mac_notification_added] = PuppetX::CiscoIOS::Utility.convert_to_boolean(instance[:mac_notification_added])
+      instance[:mac_notification_removed] = PuppetX::CiscoIOS::Utility.convert_to_boolean(instance[:mac_notification_removed])
+      instance[:link_status] = PuppetX::CiscoIOS::Utility.convert_to_boolean(instance[:link_status])
+      instance[:ip_dhcp_snooping_trust] = PuppetX::CiscoIOS::Utility.convert_to_boolean(instance[:ip_dhcp_snooping_trust])
+      instance[:flowcontrol_receive] = 'off' unless instance[:flowcontrol_receive]
+      instance[:ip_dhcp_snooping_limit] = false unless instance[:ip_dhcp_snooping_limit]
+      instance = Puppet::Provider::IosInterface::CiscoIos.clean_logging_event(instance)
+      instance = Puppet::Provider::IosInterface::CiscoIos.boolean_link_status(instance)
```

Comment:
If all this is doing is `true` or `false` then `instance[:logging_event_link_status] = instance[:logging_event_link_status] ? true : false` achieves the same

Reply:
Changed